### PR TITLE
Convert article-aside-adverts.spec to TypeScript

### PR DESCRIPTION
**Pull request by zekehuntergreen**
## What does this change?
- convert `article-aside-adverts.spec` to TypeScript
- simplify tests by using await / async rather than passing a listener to mediator which asserts the expected behaviour

## Does this change need to be reproduced in dotcom-rendering ?

- [x] No
- [ ] Yes (please indicate your plans for DCR Implementation)

## What is the value of this and can you measure success?
makes commercial modules easier to maintain and work with

### CODE REVIEW

File: static/src/javascripts/projects/commercial/modules/article-aside-adverts.spec.ts [hunk 1]
```diff
@@ -45,38 +49,42 @@ describe('Standard Article Aside Adverts', () => {
...
-		init();
+		mockMeasure(2000, 0);
+		fakeMediator.once(
+			'page:defaultcommercial:right',
+			(adSlot: HTMLElement) => {
+				expect(adSlot.classList).toContain('js-sticky-mpu');
+				expect(adSlot.getAttribute('data-mobile')).toBe(
+					'1,1|2,2|300,250|300,274|300,600|fluid',
+				);
+				done();
```
Comment (arelra) [inline static/src/javascripts/projects/commercial/modules/article-aside-adverts.spec.ts:69]: Not your code but Is it possible to change the uses of `done()` to `async () => {` and then `await` for any promise calls.

To my eyes it reads a bit easier.

Examples here: https://jestjs.io/docs/tutorial-async

Comment (zekehuntergreen) [reply static/src/javascripts/projects/commercial/modules/article-aside-adverts.spec.ts:69]: async / await seem more straightforward to me too but I'm struggling to figure out how to use them here.
How can we define `adSlot` if not as a parameter of `once`'s listener function?

Comment (arelra) [reply static/src/javascripts/projects/commercial/modules/article-aside-adverts.spec.ts:69]: Assuming that `init()` has its promises properly chained I _think_ you should be able to:

1. remove the `done()` function,
1. `await init()`
1. make the test function `async` so Jest knows to await the function

This test passes an actual function which internally wraps test assertions.

However usually for callbacks you'd want to pass a mock function i.e. `jest.fn()` in place of the callback, then you can assert later on what arguments it was called with.

Comment (zekehuntergreen) [reply static/src/javascripts/projects/commercial/modules/article-aside-adverts.spec.ts]: Thanks, that makes sense - hows that?

**Review by arelra — APPROVED** [review]
Nice simplification of the tests! 👍

File: static/src/javascripts/projects/commercial/modules/article-aside-adverts.spec.ts [hunk 2]
```diff
@@ -152,17 +144,15 @@ describe('Immersive Article (no immersive elements) Aside Adverts', () => {
...
-		fakeMediator.once('page:defaultcommercial:right', (adSlot) => {
-			expect(adSlot.classList).toContain('js-sticky-mpu');
-			expect(adSlot.getAttribute('data-mobile')).toBe(
-				'1,1|2,2|300,250|300,274|300,600|fluid',
-			);
-			done();
-		});
-		init();
+	it('should have the correct size mappings and classes (leaves it untouched)', async () => {
+		mockMeasure(900000, 0);
```
Comment (mxdvl) [inline static/src/javascripts/projects/commercial/modules/article-aside-adverts.spec.ts:148]: All them zeroes!

```suggestion
		mockMeasure(900_000, 0);
```

**Review by mxdvl — APPROVED** [review]

**Pull request merged**